tests/test_restrict_hosts.py

```diff
@@ -0,0 +1,235 @@
+    ])
+
+
+def test_marker_help_message(testdir):
+    result = testdir.runpytest(
+        '--markers',
+    )
+    result.stdout.fnmatch_lines([
+        '@pytest.mark.restrict_hosts([[]hosts[]]): Restrict socket connection to defined list of hosts',
+    ])
+
+
+def test_default_connect_enabled(assert_connect):
+    assert_connect(True)
+
+
+def test_single_cli_arg_connect_enabled(assert_connect):
+    assert_connect(True, cli_arg=localhost)
+
+
+def test_multiple_cli_arg_connect_enabled(assert_connect):
+    assert_connect(True, cli_arg=localhost + ',1.2.3.4')
+
+
+def test_single_mark_arg_connect_enabled(assert_connect):
+    assert_connect(True, mark_arg=localhost)
+
+
+def test_multiple_mark_arg_csv_connect_enabled(assert_connect):
+    assert_connect(True, mark_arg=localhost + ',1.2.3.4')
+
+
+def test_multiple_mark_arg_list_connect_enabled(assert_connect):
+    assert_connect(True, mark_arg=[localhost, '1.2.3.4'])
+
+
+def test_mark_cli_conflict_mark_wins_connect_enabled(assert_connect):
+    assert_connect(True, mark_arg=[localhost], cli_arg='1.2.3.4')
+
+
+def test_single_cli_arg_connect_disabled(assert_connect):
+    assert_connect(False, cli_arg='1.2.3.4')
+
+
+def test_multiple_cli_arg_connect_disabled(assert_connect):
+    assert_connect(False, cli_arg='5.6.7.8,1.2.3.4')
+
+
+def test_single_mark_arg_connect_disabled(assert_connect):
+    assert_connect(False, mark_arg='1.2.3.4')
+
+
+def test_multiple_mark_arg_csv_connect_disabled(assert_connect):
+    assert_connect(False, mark_arg='5.6.7.8,1.2.3.4')
+
+
+def test_multiple_mark_arg_list_connect_disabled(assert_connect):
+    assert_connect(False, mark_arg=['5.6.7.8', '1.2.3.4'])
+
+
+def test_mark_cli_conflict_mark_wins_connect_disabled(assert_connect):
+    assert_connect(False, mark_arg=['1.2.3.4'], cli_arg=localhost)
+
+
+def test_default_urllib_succeeds_by_default(assert_connect):
+    assert_connect(True, code_template=urlopen_code_template)
+
+
+def test_single_cli_arg_urlopen_enabled(assert_connect):
+    assert_connect(True, cli_arg=localhost + ',1.2.3.4', code_template=urlopen_code_template)
+
+
+def test_single_mark_arg_urlopen_enabled(assert_connect):
+    assert_connect(True, mark_arg=[localhost, '1.2.3.4'], code_template=urlopen_code_template)
+
+
+def test_global_restrict_via_config_fail(testdir):
+    testdir.makepyfile("""
+        import socket
+
+        def test_global_restrict_via_config_fail():
+            socket.socket().connect(('127.0.0.1', 80))
+    """)
+    testdir.makeini("""
+        [pytest]
+        addopts = --restrict-hosts=2.2.2.2
+    """)
+    result = testdir.runpytest("--verbose")
+    result.assert_outcomes(0, 0, 1)
+    assert_host_blocked(result, '127.0.0.1')
+
+
+def test_global_restrict_via_config_pass(testdir, httpserver):
+    httpserver.serve_content('ok')
+    test_url = urlparse(httpserver.url)
+    testdir.makepyfile("""
+        import socket
+
+        def test_global_restrict_via_config_fail():
```

Comment:
While the name of the internal method isn't strictly used, I've tried to keep it reflecting the outer test desire - in this case it's expecting to pass, not fail.

Reply:
This is a typo